test/e2e/framework/util.go

```diff
@@ -2405,10 +2405,17 @@ func WaitForStableCluster(c clientset.Interface, masterNodes sets.String) int {
 
 	}
 	allPods.Items = currentPods
+	systemPods, err := c.CoreV1().Pods(metav1.NamespaceSystem).List(metav1.ListOptions{})
+	ExpectNoError(err)
 	scheduledPods, currentlyNotScheduledPods := e2epod.GetPodsScheduled(masterNodes, allPods)
-	for len(currentlyNotScheduledPods) != 0 {
+
+	// Wait for all pending pods to be scheduled, and for only system pods to be scheduled
+	for len(currentlyNotScheduledPods) != 0 && len(scheduledPods) - len (systemPods.Items) != 0 {
```

Comment:
use a variable, something like ```numNonSystemPods := len(scheduledPods) - len (systemPods.Items)```

Reply:
Is it possible that some system Pods are not scheduled? For example:

- currentlyNotScheduledPods: 2 regular Pods, 1 system Pods
- scheduledPods: 1 regular Pod, 2 system Pod
- systemPods: 3 Pods (1 not scheduled and 2 scheduled)

In this case, len(scheduledPods) - len (systemPods.Items) == 0.

Reply:
@Huang-Wei true, this probably isn't a great check. Maybe something more explicit like a separate loop that just waits for len(allNamespacesPods) == len(systemPods)?

Reply:
you can do this:

```
scheduledSystemPods, currentlyNotScheduledSystemPods = e2epod.GetPodsScheduled(masterNodes, systemPods)
```

and  the check would be```currentlyNotScheduledSystemPods != 0 || len(systemPods.Items) != len(allPods.Items)```

Reply:
@ahg-g updated with this check